dangerzone/document.py

```diff
@@ -82,3 +101,33 @@ def set_default_output_filename(self) -> None:
         self.output_filename = (
             f"{os.path.splitext(self.input_filename)[0]}{SAFE_EXTENSION}"
         )
+
+    def is_unconverted(self) -> bool:
+        return self.state is Document.STATE_UNCONVERTED
+
+    def is_converting(self) -> bool:
+        return self.state is Document.STATE_CONVERTING
+
+    def is_failed(self) -> bool:
+        return self.state is Document.STATE_FAILED
+
+    def is_safe(self) -> bool:
+        return self.state is Document.STATE_SAFE
+
+    def mark_as_converting(self) -> None:
+        log.debug(f"Marking doc {self} as 'converting'")
+        self.state = Document.STATE_CONVERTING
+
+    def mark_as_failed(self) -> None:
+        log.debug(f"Marking doc {self} as 'failed'")
+        self.state = Document.STATE_FAILED
+
+    def mark_as_safe(self) -> None:
+        log.debug(f"Marking doc {self} as 'safe'")
+        self.state = Document.STATE_SAFE
+
+    def get_short_id(self) -> str:
+        return str(self.id)[0:6]
+
+    def __str__(self) -> str:
+        return self.get_short_id()
```

Comment:
I wouldn't suggest overriding `__str__()` to print a random value, as this will make debugging weirder. I would leave `__str__` as is, turn `self.get_short_id()` into a property (`self.short_id`) and use it where necessary.